### ISSUE TITLE
[BUG] Strange "h4039" element on default landing page

### ISSUE THREAD

**Issue by joewiz** [eXist-db/exist-markdown#8]
**Describe the bug**

When loading the landing page (`/main.md`), the generated HTML has a strange `<h4039>` element inside the `body/section`:

```html
<body class="container">
    <body>
        <section>
            <h4039># Supported Markdown syntax
```

Markdown within this element is not further processed or transformed into HTML.

**Expected behavior**

The page should contain valid HTML.

**To Reproduce**

Install app, load http://localhost:8080/exist/apps/markdown.

**Context (please always complete the following information):**
 - OS: macOS 11.4
 - eXist-db Version: eXist 5.3.0-SNAPSHOT e371efd9987a9a2f4414839c7bf1dbc20107b6d1 20210604033555
 - Java Version: OpenJDK 1.8.0_292-b10 (liberica-jdk8-full) 
-  App Version: 0.6 (both installed from public-repo, and built from current master)

**Additional context**
- How is eXist-db installed? built from source
- Any custom changes in e.g. `conf.xml`? none

**Comment by joewiz** [comment]
Further analysis suggests that the inline parsing is working, but block parsing is failing on the first heading. Here's the compete transformation of the [test.md](https://raw.githubusercontent.com/eXist-db/exist-markdown/master/test.md) file which is rendered when requested via the http://localhost:8080/exist/apps/markdown/test.md endpoint:

```xml
<html>
    <head>
        <title>Markdown Parser Test</title>
        <style type="text/css">
                td.left { text-align: left; }
                td.right { text-align: right; }
                td.center { text-align: center; }</style>
    </head>
    <body class="container">
        <body>
            <section>
                <h4039># Supported Markdown syntax
                    
                    The parser extends the <a href="#">No link definition found for id 3</a> proposal with fenced code blocks and tables. These are additional features found in <a href="#">No link definition found for id 2</a>.
                    
                    ## Paragraphs
                    Paragraphs are separated from following blocks by a blank line. 
                    A single line break does <strong>not</strong> start a new paragraph.
                    
                    Lorem ipsum dolor sit amet, consectetur adipiscing elit. Curabitur nec lobortis magna. Fusce vestibulum felis a eros suscipit mattis. Pellentesque sit amet enim libero. Sed sed tempus nibh. Ut pellentesque quam ac bibendum iaculis. Suspendisse <strong>vitae</strong> interdum risus, convallis auctor urna. Mauris vel sapien ut sapien mollis rhoncus non a nibh. Nullam vulputate consequat purus, ut varius justo ornare vel. Etiam ornare diam at velit varius volutpat. Mauris vel luctus mi, at fermentum purus. <em>Vestibulum ante ipsum</em> primis in faucibus orci luctus et ultrices posuere cubilia Curae; Cras lobortis est dolor, et tristique lorem egestas vitae. Sed feugiat dictum nunc. Nullam ultricies vehicula aliquam. Cras felis ante, ultrices sed lacinia et, pharetra in tellus. Vivamus scelerisque ut mi a dapibus.
                    
                    ## Code
                    
                    To format inline code snippets, surround them with a single backtick: <code>request:get-parameter()</code>. Use two 
                    backticks to allow one backtick inside: <code> `ls` </code>.
                    
                    ## Lists
                    
                    ### Simple list:
                    
                    <em> Buy milk
                    </em> Drink it
                    <em> Be happy
                        
                        ### Nested list:
                        
                        1. One
                        2. Two
                    </em> A nested list item
                    <em> in an unordered list.
                        3. Four
                        
                        ### Task List
                        
                    </em> [x] write documentation
                    * [ ] create tests
                    
                    ### Quotes
                    
                    &gt; Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim 
                    &gt; veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate 
                    &gt; velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit 
                    &gt; anim id est laborum.
                    
                    ## Links
                    
                    Links can be specified directly or by reference.
                    
                    This <a href="#">No link definition found for id 1</a> references a link definition given at the end of the document ! And here's a direct link to the eXist <a href="http://exist-db.org/exist/apps/docs" title="eXist-db Documentation">documentation</a>.
                    
                    ## Images
                    
                    <img src="http://exist-db.org/exist/apps/homepage/resources/img/existdb.gif" alt="eXist-db Logo" title="Our Logo">
                        
                        Image linked through reference: <img src="#" alt="No link definition found for id glasses">.
                            
                            ## Labels
                            
                            <em> <span itemprop="customer">eXist Solutions</span>, <span itemprop="customer">foo enterprise</span>
                            </em> <span itemprop="important">important</span>
                            
                            ## Code Blocks
                            
                            ```xquery
                            for $i in 1 to 10
                            return
                            &lt;li&gt;{$i <em> 2}&lt;/li&gt;
                                ```
                                
                                ## Table
                                
                                | Tables        | Are           | Cool  |
                                | ------------- |:-------------:| -----:|
                                | col 3 is      | right-aligned | $1600 |
                                | col 2 is      | </em>centered<strong>  |   $12 |
                                    | zebra stripes | are neat      |    $1 |
                                    
                                    simple table | column1 | column2
                                    
                                    ## HTML Blocks
                                    
                                    ``<code>xml
                                        &lt;figure&gt;
                                        &lt;img src="http://exist-db.org/exist/apps/homepage/resources/img/existdb.gif"/&gt;
                                        &lt;/figure&gt;
                                    </code>``
                                    
                                    is rendered as:
                                    
                                    &lt;figure&gt;
                                    &lt;img src="http://exist-db.org/exist/apps/homepage/resources/img/existdb.gif"/&gt;
                                    &lt;/figure&gt;
                                    
                                    HTML block containing markdown:
                                    
                                    &lt;div class="row"&gt;
                                    &lt;div class="col-md-6"&gt;
                                    First column in </strong>two column layout**.
                            
                            Second paragraph.
                            &lt;/div&gt;
                            &lt;div class="col-md-6"&gt;
                            Second column in two column layout.
                            &lt;/div&gt;
                            &lt;/div&gt;
                            
                            ## Inline HTML
                            
                            A <span style="color: red;">paragraph <span style="color: green;">containing</span></span> some  <code>HTML</code>.
                            
                            # TEI output
                            
                            Besides producing HTML, the module can also transform <a href="?mode=tei">markdown into TEI</a>. To test, just append <code>?mode=tei</code>
                            to the URL. Other output formats can be supported as well by adding a simple configuration, see <a href="https://github.com/wolfgangmm/exist-markdown/blob/master/content/tei-config.xql">tei-config.xql</a>.
                            
                            [1]: http://exist-db.org "eXist-db homepage"
                            [2]: https://help.github.com/articles/github-flavored-markdown
                            [3]: http://daringfireball.net/projects/markdown/syntax
                            [glasses]: http://exist-db.org/exist/apps/homepage/resources/img/book-cover.gif "Documentation"
                </h4039>
            </section>
        </body>
    </body>
</html>
```

**Comment by joewiz** [comment]
Closed by #12.
